### PR TITLE
[FEAT]: Workflow Response: Routes, Handler & Processing Utils

### PR DESCRIPTION
**Pull request by MuhammadUmer44**
### Describe the chnages:
- Implement  workflow responses functionality.

closes: #1941

closes: #1942

closes: #1943

## Issue ticket number and link:
- **Ticket Number:** [ 1941, 1942, 1943]
- **Link:** [ https://github.com/stakwork/sphinx-tribes/issues/1941, https://github.com/stakwork/sphinx-tribes/issues/1942, https://github.com/stakwork/sphinx-tribes/issues/1943]

### Evidence:

WorkflowResponseTest

Route-test

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome
- [x] New feature (non-breaking change which adds functionality)
- [x] I have provided a screenshot of changes in my PR

### CODE REVIEW

**Comment by MuhammadUmer44** [comment]
Hi @humansinstitute, Please review this PR.

**Comment by humansinstitute** [comment]
Seems to me that we're missing the util functions here to follow the format in the design? 

https://github.com/stakwork/sphinx-tribes/issues/1943

**Comment by MuhammadUmer44** [comment]
> Seems to me that we're missing the util functions here to follow the format in the design?
> 
> #1943



> Seems to me that we're missing the util functions here to follow the format in the design?
> 
> #1943

Hi @humansinstitute, @tobi-bams, @elraphty,  

The processing functionality is already implemented in our current codebase. In `HandleWorkflowResponse`, we

`1- Look up processing requirements:`

https://github.com/stakwork/sphinx-tribes/blob/6d75668bb2b040b1675f005106bf1172f2081e6f/handlers/workflow.go#L93

`2- Set status to PENDING if processing is required:`

https://github.com/stakwork/sphinx-tribes/blob/6d75668bb2b040b1675f005106bf1172f2081e6f/handlers/workflow.go#L101C2-L103C3


All the required fields (Source, RequestID, Action, WorkflowID, RequestData, ResponseData) are already handled in our WfRequest struct.
The only difference is that instead of having this in a utils file, it's integrated into our handler. The functionality matches the design requirements, just with a different structure optimized for our async workflow.

- Uses GetProcessingMapByKey to check processing requirements
- Sets status to PENDING if processing needed
- Stores response data for async processing

@tobi-bams, @elraphty, could you please review this PR?

**Comment by humansinstitute** [comment]
@tobi-bams could you take a look at this and approve?

File: handlers/workflow.go [hunk 1]
```diff
@@ -17,14 +19,103 @@ func NewWorkFlowHandler(database db.Database) *workflowHandler {
...
+		http.Error(w, "Invalid response format", http.StatusBadRequest)
+		return
+	}
+
+	if response.RequestID == "" {
+		http.Error(w, "Request ID is required", http.StatusBadRequest)
+		return
+	}
+
+	switch response.Status {
```
Comment (tobi-bams) [inline handlers/workflow.go:75]: We don't need this `switch` block

File: handlers/workflow.go [hunk 2]
```diff
@@ -17,14 +19,103 @@ func NewWorkFlowHandler(database db.Database) *workflowHandler {
...
+		return
+	}
+
+	processingMap, err := wh.db.GetProcessingMapByKey(request.Source, request.Action)
+	if err != nil {
+		http.Error(w, "Failed to check processing requirements", http.StatusInternalServerError)
+		return
+	}
+
+	status := response.Status
```
Comment (tobi-bams) [inline handlers/workflow.go:99]: The status here should be completed automatically

**Review by tobi-bams — APPROVED** [review]

**Pull request merged**